ngraph/frontend/paddlepaddle/src/model.cpp

```diff
@@ -502,6 +475,250 @@ namespace ngraph
             m_tensor_values[name] = constant;
         }
 
+        Place::Ptr InputModelPDPD::InputModelPDPDImpl::addOutput(Place::Ptr place)
+        {
+            auto tensor_place = pdpd::castToTensorPlace(place);
+            if (!tensor_place->is_output())
+            {
+                m_outputs.push_back(place);
+                return tensor_place;
+            }
+            FRONT_END_THROW("Place is already output.");
+        }
+
+        void InputModelPDPD::InputModelPDPDImpl::removeOutput(Place::Ptr place)
+        {
+            auto tensor_place = pdpd::castToTensorPlace(place);
+            m_outputs.erase(std::remove(m_outputs.begin(), m_outputs.end(), tensor_place),
+                            m_outputs.end());
+            m_graph_changed = true;
+        }
+
+        void InputModelPDPD::InputModelPDPDImpl::cutAndAddNewOutput(
+            Place::Ptr place, const std::string& new_name_optional)
+        {
+            auto tensor_place = pdpd::castToTensorPlace(place);
+            if (tensor_place)
+            {
+                std::vector<std::shared_ptr<TensorPlacePDPD>> new_tensors;
+
+                // Find outputs that are connected to the cut place. These target outputs should be
+                // cut off.
+                traverse_down({tensor_place}, nullptr, &new_tensors);
+                std::vector<std::shared_ptr<TensorPlacePDPD>> target_out_tensor;
+                for (const auto& tensor : new_tensors)
+                {
+                    if (tensor->is_output())
+                    {
+                        target_out_tensor.push_back(tensor);
+                    }
+                }
+                m_outputs.push_back(tensor_place);
+
+                // If some target outputs still connected with the model inputs, the selected cut
+                // place is incorrect.
+                new_tensors.clear();
+                traverse_down(m_inputs, nullptr, &new_tensors);
+                std::vector<Place::Ptr> new_outputs;
+                for (const auto& tensor : new_tensors)
+                {
+                    if (tensor->is_output())
+                    {
+                        FRONT_END_GENERAL_CHECK(std::find(target_out_tensor.begin(),
+                                                          target_out_tensor.end(),
+                                                          tensor) != target_out_tensor.end(),
+                                                "Incorrect Place for cutting.");
```

Comment:
Can it be more informative, i.e. add output's tensor's name which depends on new output?

Reply:
done